Modules/CustomerHistory/Frames.lua

```diff
@@ -24,15 +24,46 @@ function CraftSim.CUSTOMER_HISTORY.FRAMES:Init()
     local function createContent(frame)
         self.frame:Hide()
         self.frame.content.customerDropdown = CraftSim.GGUI.Dropdown({
-            parent=self.frame.content, anchorParent=self.frame.title.frame, anchorA="TOP", anchorB="TOP", offsetY=-30, width=170, offsetX=-170,
+            parent=self.frame.content, anchorParent=self.frame.title.frame, anchorA="TOP", anchorB="TOP", width=170, offsetX=-170, offsetY=-30,
             initialValue=nil,
             initialLabel="",
             label=CraftSim.LOCAL:GetText(CraftSim.CONST.TEXT.CUSTOMER_HISTORY_DROPDOWN_LABEL),
             initialData={},
             clickCallback=function (_, _, item) CraftSim.CUSTOMER_HISTORY.FRAMES:SetCustomer(item) end
         })
+        self.frame.content.deleteButton = CraftSim.GGUI.Button({
+            parent=frame.content,anchorParent=self.frame.content.customerDropdown.frame, anchorA="LEFT", anchorB="RIGHT", offsetX=0, offsetY=2,
+            label=CraftSim.LOCAL:GetText(CraftSim.CONST.TEXT.CUSTOMER_HISTORY_DELETE_BUTTON), sizeX=15, sizeY=20, adjustWidth=true,
+            clickCallback=function ()
+                CraftSim.CUSTOMER_HISTORY.db.realm[CraftSim.CUSTOMER_HISTORY.db.realm.lastCustomer] = nil
+                self:ResetDropdown()
+                self.frame.content.messageBox:Clear()
+                self:LoadTotalTip()
+                CraftSim.CUSTOMER_HISTORY.db.realm.lastCustomer = nil
+                self.frame.content.deleteButton.frame:SetEnabled(false)
+            end,
+        })
+        self.frame.content.deleteButton.frame:SetEnabled(false)
+
+        self.frame.content.filterTextbox = CraftSim.FRAME:CreateInput("Search", self.frame.content, self.frame.content.customerDropdown.frame, "TOP", "BOTTOM", 0, 0, 170, 25,
+        "",
+        function(textInput, userInput)
+            local input = textInput:GetText():gsub("^%s*(.-)%s*$", "%1") or ""
+            if input ~= "" then
+                self:ResetDropdown(CraftSim.CUSTOMER_HISTORY.db.realm.lastCustomer, input)
```

Comment:
From my tests, userInput is always true, probably because it’s always shown and enabled when using it
But I may have made a mistake, I’ll take a look later when I get home

Reply:
Ah right, that was my initial comment, but I actually was more focused on the shown state than the dropdown content. Good catch.

Reply:
No problem 😄 
I won’t be home for a few hours, if you’re willing to you’re more than welcome to make the change !